impl/src/error.rs

```diff
@@ -18,7 +18,7 @@ pub fn expand(
     let state = State::with_attr_params(
         input,
         trait_name,
-        quote! { ::std::error },
+        quote!{ ::derive_more::__private::Error },
```

Comment:
> We can bring the trait in scope together with the macro. This may be handy for much cases. And we can do this for all the traits (in a separate PR, surely).

Yeah, I think that's probably a good idea. Because of this change I'm working on adding the following line to all our tests:
```
#![cfg_attr(not(feature = "std"), no_std)]
```
And automatically importing the traits, possibly from `alloc` would be quite useful.

Reply:
I'll leave it as __private here and we can make it public in the follow up PR where we do this for all traits.